### PR TITLE
Upgrade to webpack 5

### LINKED ISSUE #4675: Webpack 5
It has some very interesting features like module federation. But the painful part will be upgrading our existing webpack build configs to Webpack 5 first.

### PR DESCRIPTION
**Pull request by plourenco**
Closes #4675

### CODE REVIEW

File: indico/web/client/js/react/components/index.js [hunk 1]
```diff
@@ -5,6 +5,14 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
+export {
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/index.js:8]: Why was this move needed? (Requiring a particular export order seems rather fragile!)

File: webpack.config.babel.js [hunk 2]
```diff
@@ -150,8 +151,5 @@ export default env => {
     resolve: {
       alias: extraResolveAliases,
     },
-    output: {
-      jsonpFunction: 'coreJsonp',
-    },
```
Comment (ThiefMaster) [inline webpack.config.babel.js:155]: I think removing this may break plugins; those are separate webpack builds and thus need separate callbacks.

Comment (plourenco) [reply webpack.config.babel.js:155]: What was the cause for adding it?

File: indico/web/client/js/react/components/index.js (hunk 1 above)
Comment (plourenco) [reply indico/web/client/js/react/components/index.js:8]: I forgot to comment this in one of the commits. The cause is this exact webpack-contrib/mini-css-extract-plugin#382
A clearer explanation is also [here](https://github.com/facebook/create-react-app/issues/5372#issuecomment-685932009).

It's possibly safe to ignore these warnings, however, this was the only change required to silence then alll.

File: webpack.config.babel.js (hunk 2 above)
Comment (ThiefMaster) [reply webpack.config.babel.js:155]: @pferreir may remember the details, otherwise the git blame and then the commit/PR adding it may also help..

Comment (pferreir) [reply webpack.config.babel.js:155]: https://github.com/indico/indico/commit/902c217299d89c927ac9df961bff4c1946dfc623

Comment (pferreir) [reply webpack.config.babel.js:155]: > This keeps the plugin environment from colliding with the core in
production builds.

File: webpack/base.js [hunk 3]
```diff
@@ -284,14 +282,6 @@ export function webpackDefaults(env, config, bundles, isPlugin = false) {
       new MiniCssExtractPlugin({
         filename: 'css/[name].[contenthash:8].css',
       }),
-      new ProgressBarPlugin({
```
[the remark this answers is not in the document]
Comment (plourenco) [reply webpack/base.js:306]: It's still supported, yes. I'm suggesting (in the latest commit) replacing it with webpack's `--progress` and `--profile` flags which are more informative. But I can always revert this.

[the remark this answers is not in the document]
Comment (ThiefMaster) [reply webpack/base.js:306]: i just tested it and the new progress display is nice. :) ("sending code to the moon and back" was fun though :p)

**Comment by ThiefMaster** [comment]
Room booking (e.g. the list of rooms) is broken due to "process is not defined". Likely this issue: https://stackoverflow.com/a/65018686/298479

File: webpack.config.babel.js (hunk 2 above)
Comment (ThiefMaster) [reply webpack.config.babel.js:155]: https://webpack.js.org/blog/2020-10-10-webpack-5-release/#automatic-unique-naming

So I guess this is no longer needed :)

**Comment by plourenco** [comment]
> Room booking (e.g. the list of rooms) is broken due to "process is not defined". Likely this issue: https://stackoverflow.com/a/65018686/298479

Thanks, added one for path but forgot this one.

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**